assets/src/components/groupedAutocomplete.tsx

```diff
@@ -0,0 +1,731 @@
+          // Delete the cursor state if the user's focus exits the autocomplete
+          // control.
+          if (event.currentTarget.contains(event.relatedTarget)) {
+            return
+          }
+
+          updateCursorLocation(CursorLocationAction.DeleteCursor)
+        }}
+        onKeyDown={(e) => {
+          // Handle cursor movement by keyboard.
+          // If the key is not a movement key, allow event to bubble up.
+          switch (e.key) {
+            case "ArrowDown":
+              updateCursorLocation(CursorLocationAction.MoveNext)
+              break
+            case "ArrowUp":
+              updateCursorLocation(CursorLocationAction.MovePrev)
+              break
+            case "Home":
+              updateCursorLocation(CursorLocationAction.MoveToStart)
+              break
+            case "End":
+              updateCursorLocation(CursorLocationAction.MoveToEnd)
+              break
+            default:
+              return
+          }
+          e.preventDefault()
+          e.stopPropagation()
+        }}
+        aria-labelledby={listHeadingId}
+        tabIndex={-1}
+      >
+        {optionGroups.map(({ group: { title, options } }, groupIndex) => (
+          <GroupOptionList key={groupIndex} heading={title}>
+            {options.map(
+              ({ option: { label, onSelectOption } }, optionIndex) => {
+                const selected =
+                  cursorLocation &&
+                  cursorLocation.group === groupIndex &&
+                  cursorLocation.option === optionIndex
+                return (
+                  <li
+                    key={optionIndex}
+                    className="c-autocomplete__option"
+                    role="option"
+                    aria-selected={selected}
+                    onFocus={(e) => {
+                      // Set cursor to this location if focused.
+                      !selected &&
+                        setCursorLocation({
+                          group: groupIndex,
+                          option: optionIndex,
+                        })
+                      e.stopPropagation()
+                    }}
+                    onClick={onSelectOption}
+                    onKeyDown={(e) => {
+                      // Fire `onSelectOption` if enter is pressed
+                      if (e.key != "Enter") {
+                        return
+                      }
+                      e.preventDefault()
+                      e.stopPropagation()
+                      onSelectOption?.(e)
+                    }}
+                    // If this element is selected by the cursor,
+                    // set document focus to this element when mounted.
+                    ref={(selected || null) && ((r) => r?.focus())}
+                    // Allow user to tab into the first option and out of
+                    // the autocomplete control on next `Tab` key.
+                    // > Composite controls should have one tab stop
```

Comment:
> Composite controls should have one tab stop

question: What is this referencing?

Reply:
I was certain this was in the WCAG or something but I must've unintentionally paraphrased where I read this from because I can't find it. Will revise.

Reply:
I thinking of removing this altogether and let arrow keys be the primary method of entering the control, because [WebAIM's documentation on Keyboard Accessibility](https://webaim.org/techniques/keyboard/) says that 

> Tab panel
> - `Tab` - once to navigate into the group of tabs and once to navigate out of the group of tabs
> - `↑`/`↓` or `←`/`→` - choose and activate previous/next tab.

but the autocomplete section states that arrow keys are the way to navigate into and out of the composite control.

> Autocomplete
> - Type to begin filtering
> - `↑`/`↓` - navigate to an option
> - `Enter` - select an option

But I wonder if having it be tab-able is something _our_ users would want? It's probably too early to say so maybe removing this is a better option till usability testing.

Reply:
1242d36129eb26966573a1ae2d2b2906d3bf3519